lms/djangoapps/program_enrollments/api/v1/views.py

```diff
@@ -124,3 +133,222 @@ def get(self, request, program_key=None):
         paginated_enrollments = self.paginate_queryset(enrollments)
         serializer = ProgramEnrollmentListSerializer(paginated_enrollments, many=True)
         return self.get_paginated_response(serializer.data)
+
+
+class ProgramSpecificViewMixin(object):
+    """
+    A mixin for views that operate on or within a specific program.
+    """
+    def __init__(self, *args, **kwargs):
+        super(ProgramSpecificViewMixin, self).__init__(*args, **kwargs)
+        self._program = None
```

Comment:
Why have this local class level cache? `get_programs` is already getting data from the cache. You are not improving the performance by having this variable.

Reply:
you're right, but I heavily based this off of the equivalent code in registrar, and that's how it's being done there